src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+				dX = Math.Max(0, dX);
+				dY = Math.Max(0, dY);
+				clipSize.Width = Math.Max(0, toolTipRect.Right - dX);
+				clipSize.Height = Math.Max(0, toolTipRect.Bottom - dY);
+				PerformClipping(clipSize);
+			}
+		}
+
+		private void MovePointToPointerToolTipShowPosition(
+			ref Point point,
+			PlacementMode placement)
+		{
+			// If the point is inside the placement rect, move it out of the placement rect.
+			var placementRect = GetPlacementRectInWindowCoordinates();
+
+			if (!placementRect.IsEmpty && placementRect.Contains(point))
+			{
+				switch (placement)
+				{
+					case PlacementMode.Left:
+						point.X = placementRect.X;
+						break;
+					case PlacementMode.Right:
+						point.X = placementRect.X + placementRect.Width;
+						break;
+					case PlacementMode.Top:
+					case PlacementMode.Mouse:
+						point.Y = placementRect.Y;
+						break;
+					case PlacementMode.Bottom:
+						point.Y = placementRect.Y + placementRect.Height;
+						break;
+				}
+			}
+		}
+
+		private void MovePointToPointerToolTipShowPosition(
+			ref double left,
+			ref double top,
+			PlacementMode placement)
+		{
+			var point = new Point(left, top);
+
+			MovePointToPointerToolTipShowPosition(ref point, placement);
+
+			left = point.X;
+			top = point.Y;
+		}
+
+		private Rect GetPlacementRectInWindowCoordinates()
+		{
+			var placementRectLocal = Rect.Empty;
+
+			if (PlacementRect is Rect placementRect)
+			{
+				placementRectLocal = placementRect;
+
+				if (Target != null)
+				{
+					var tr = Target.TransformToVisual(null);
+					tr.TransformBounds(placementRectLocal);
+				}
+			}
+
+			return placementRectLocal;
+		}
+
+		// If the owner is a TextElement, we'll get its bounding rect and use that as our placement target rect.
+		// Otherwise, we'll use the default rect derived from the target.
+		private void PerformPlacementInternal()
+		{
+			if (_owner is TextElement textElement)
+			{
+				// UNO TODO
+				//        CCoreServices* pCore = static_cast<CCoreServices*>(DXamlCore::GetCurrent()->GetHandle());
+
+				//XRECTF boundingRect;
+				//pCore->GetTextElementBoundingRect(ownerAsTextElement.Cast<TextElement>()->GetHandle(), &boundingRect);
+
+				//        RECT rectDockTo =
+				//		{
+				//			static_cast<long>(boundingRect.X),
+				//			static_cast<long>(boundingRect.Y),
+				//			static_cast<long>(boundingRect.X + boundingRect.Width),
+				//			static_cast<long>(boundingRect.Y + boundingRect.Height)
+				//		};
+				//PerformPlacement(&rectDockTo);
+			}
+			else
+			{
+				PerformPlacement();
+			}
+		}
+
+		// Handle the SizeChanged event.
+		private void OnToolTipSizeChanged(
+			object pSender,
+			SizeChangedEventArgs pArgs)
+		{
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().LogDebug($"{nameof(OnToolTipSizeChanged)} - new Size=({ActualWidth}, {ActualHeight})");
+			}
+			PerformPlacementInternal();
+		}
+
+		private void OnPopupOpened(
+				object pUnused1,
+				object pUnused2)
+		{
+			//OnOpened();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)